src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -48,73 +46,98 @@ const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleC
     get dirty() {
-      const providedValue = props.value ? props.value.id : undefined;
+      const providedValue = props.value[0] !== -1 ? props.value[0] : undefined;
       return providedValue !== this.value[0];
     },
     id: props.id,
     name: props.name,
     get value() {
-      return value ? [value.id] : [];
+      return value[0] === -1 ? [] : value;
     },
   }));
 
   useEffect(() => {
     setValue(props.value);
-  }, [props.value]);
+  }, [...props.value]);
 
   useEffect(() => {
-    if (props.value === value) return;
+    if (JSON.stringify(props.value) === JSON.stringify(value)) return;
 
     props.onChange({ target: selfRef.current });
   }, [value]);
 
+  useEffect(() => {
+    const getChoices = async () => {
+      await execute(`${API_ROOT}/custom_field_choices?for_custom_fields=${props.customFieldID}`)
+        .then(({ json, mounted }) => {
+          if (mounted) {
+            const mungedChoices = json.results.map((result) => {
+              return {
+                ...json[result.key][result.id],
+                id: result.id,
+              };
+            });
+
+            setChoices(mungedChoices);
+          }
+        })
+        .catch((error) => {
+          if (error.error && error.error.type !== 'aborted') {
+            throw error;
+          }
+        });
+    };
+
+    getChoices();
+  }, []);
+
   return (
-    <Select
-      className={props.className}
-      displayValueEvaluator={selectValue => selectValue.label}
-      errorText={props.errorText}
-      id={props.id}
-      label={props.label}
-      listOptionRefs={listOptionRefs}
-      name={props.name}
-      onChange={selectOnChangeHandler}
-      placeholder={props.placeholder}
-      readOnly={props.readOnly}
-      required={props.required}
-      value={value}
-    >
-      { listOptions() }
-    </Select>
+    <React.Fragment>
+      <Select
+        className={props.className}
+        displayValueEvaluator={selectValue => selectValue.label}
+        errorText={props.errorText}
+        id={props.id}
+        label={props.label}
+        listOptionRefs={listOptionRefs}
+        name={props.name}
+        onChange={selectOnChangeHandler}
+        placeholder={props.placeholder}
+        readOnly={props.readOnly}
+        required={props.required}
+        value={value[0] !== -1 ? choices.find(choice => choice.id === value[0]) : undefined}
+      >
+        { listOptions() }
+      </Select>
+      {choices.length === 0 &&
+        <span>Loading...</span>
+      }
+    </React.Fragment>
   );
 });
 
-const ChoiceType = PropTypes.shape({
-  id: PropTypes.number.isRequired,
-  label: PropTypes.string.isRequired,
-});
-
 CustomFieldInputSingleChoice.propTypes = {
   id: PropTypes.string.isRequired,
-  choices: PropTypes.arrayOf(ChoiceType),
   className: PropTypes.string,
+  /** Needs to be the Custom Field ID from the database for self-loading of choices */
+  customFieldID: PropTypes.number.isRequired,
```

Comment:
Curious about treating ID as two words instead of one: `customFieldId` /shrug We just have to stay consistent moving forward.

Reply:
Yeah, since `ID` is an abbreviation, I make the letters share case. I'm cool with either, but this is what I default to.